client/src/components/AuthNavbar/Navbar.tsx

```diff
@@ -118,6 +118,12 @@ export default function AuthNavbar(): JSX.Element {
     });
   }, []);
 
+  const handleLinkClick = () => {
+    setTimeout(() => {
+      localStorage.setItem('page_location', location.pathname);
+    }, 400);
+  };

```

Comment:
This works but it's unfortunately a really hacky solution to the problem, you don't need to use local storage and timeouts. A better approach would be to just remove the redirect to dashboard or move it to a more logical place like inside the login/signup components since those are the only ones with a redirect

Reply:
I have made changes. Please let me know about it.